### DISCUSSION TITLE
[BUG] This doesn't seem to work for the %run magic command

### DISCUSSION THREAD

**Discussion by jencake** [s-weigand/flake8-nb#286] [Q&A] [answered]
For variables are defined in another notebook and imported via %run. flake8_nb still gives me the variable is undefined error.

**Comment by s-weigand** [comment, marked as answer]
@jencake This is a general limitation when using `jupyter magic` syntax that manipulates the ipython kernel state at runtime and using static linters. 
The problem is that magic commands aren't regular python code, so it does not translate to the equivalent python AST that `flake8` uses to lint code.
The only way to fix this would require  `flake8` to use a different AST parser that supports `jupyter magic` syntax, which I'm not aware even exists and even if I don't see `flake8` changing to it.

[Pretty much the same issue was opened and closed for nbQA](https://github.com/nbQA-dev/nbQA/issues/793) and I have the [same suggestion](https://github.com/nbQA-dev/nbQA/issues/793#issuecomment-1405548690) as @MarcoGorelli to use `# noqa` either as comment inline or via [cell tags](https://flake8-nb.readthedocs.io/en/latest/usage.html#cell-tags).

The other way to solve this would be to convert the notebook you run with `%run` to a python file and use regular imports.
